pkg/proxy/iptables/proxier.go

```diff
@@ -1495,6 +1498,11 @@ func (proxier *Proxier) syncProxyRules(reason syncReason) {
 	}
 	proxier.portsMap = replacementPortsMap
 
+	// Update healthz timestamp if it is periodic sync.
+	if proxier.healthzServer != nil && reason == syncReasonForce {
```

Comment:
@nicksardo Moved `UpdateTimestamp` to the end of `syncProxyRules` so that it wouldn't be updated if anything fails in between, thanks!